pyteal/ast/itxn.py

```diff
@@ -200,6 +205,108 @@ def SetFields(cls, fields: Dict[TxnField, Union[Expr, List[Expr]]]) -> Expr:
         fieldsToSet = [cls.SetField(field, value) for field, value in fields.items()]
         return Seq(fieldsToSet)
 
+    @classmethod
+    def MethodCall(
+        cls,
+        *,
+        app_id: Expr,
+        method_signature: str,
+        args: list[abi.BaseType | Expr | dict[TxnField, Expr | list[Expr]]],
+        fields: dict[TxnField, Expr | list[Expr]],
+    ) -> Expr:
+
+        # Default, always need these
+        fields_to_set = [
+            cls.SetField(TxnField.type_enum, TxnType.ApplicationCall),
+            cls.SetField(TxnField.application_id, app_id),
+        ]
+
+        # We only care about the args
+        arg_type_specs: list[abi.TypeSpec]
+        arg_type_specs, _ = type_specs_from_signature(method_signature)
+
+        # Start app args with the method selector
+        app_args: list[Expr] = [MethodSignature(method_signature)]
+
+        # Transactions are not included in the App Call
+        txns_to_pass: list[Expr] = []
+
+        # Reference Types are treated specially
+        accts: list[Expr] = []
+        apps: list[Expr] = []
+        assets: list[Expr] = []
+
+        for idx, arg_ts in enumerate(arg_type_specs):
+            arg = args[idx]
+
+            if arg_ts in abi.TransactionTypeSpecs:
+                if not isinstance(arg, dict):
+                    raise TealTypeError(arg, dict[TxnField, Expr | list[Expr]])
```

Comment:
I'd also like to see more type checking here. We should error if you pass in the wrong type of transaction.

Checking if the dict has a `TxnField.type_enum` key and its value adheres to one of the following might work:
* `value` is an `Int` with the right value
* OR `value` is an `IntEnum` with the right string value

Since we're validating a runtime expression, this won't work in all cases. You could do something strange where the type of the txn isn't a constant, but I'd argue that's not an important scenario to support since it seems reasonable to expect inner txn types to be known at compile time.